Back-To-School/Assets/BackToSchool/Scripts/Enemies/EnemySpawner.cs

```diff
@@ -12,84 +11,98 @@ public class EnemySpawner : MonoBehaviour
     {
         public Action<int> EnemyDied;
 
-        [SerializeField] private Enemy _enemyPrefab;
-
-        [SerializeField] private float _minXpos = -49f;
-        [SerializeField] private float _maxXpos = 49f;
-        [SerializeField] private float _minZpos = -49f;
-        [SerializeField] private float _maxZpos = 49f;
-
-        [SerializeField] private float _spawnInterval = 1f;
+        [SerializeField] private EnemyWarrior _enemyWarriorPrefab;
+        [SerializeField] private EnemyShaman _enemyShamanPrefab;
         [SerializeField] private float _maxRangeToPlayer = 10f;
-        [SerializeField] private int _maxEnemies = 3;
+        [SerializeField] private float _spawnInterval = 1f;
+        [SerializeField] private int _maxEnemies = 6;
 
-        private List<Enemy> _enemies = new List<Enemy>();
+        private List<BaseEnemy> _enemies = new List<BaseEnemy>();
         private GameObject _target;
         private Vector3 _enemyPos = Vector3.zero;
 
         private float _xPos;
         private float _yPos = 0f;
         private float _zPos;
         private float _timer;
-        private int _currentNumberOfEnemies;
+        private int _halfOfEnemies;
```

Comment:
Maybe make just separate limit for each type? It would be much simpler to understand/control and seems like adding more enemy types would broke such logic